app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/common/Presenter.kt

```diff
@@ -0,0 +1,17 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common
+
+import io.reactivex.disposables.Disposable
+
+abstract class Presenter {
+
+    var subscriptions: List<Disposable> = emptyList()
+
+    // Here view might not be loaded yet
+    open fun onCreate() {}
+
+    open fun onStart() {}
+
+    open fun onDestroy() {
+        subscriptions.forEach { it.dispose() }
```

Comment:
use this instead http://reactivex.io/RxJava/2.x/javadoc/io/reactivex/disposables/CompositeDisposable.html